x-pack/plugins/maps_legacy_licensing/public/plugin.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignore
+import { CoreSetup, CoreStart, Plugin } from 'kibana/public';
+
+/**
+ * These are the interfaces with your public contracts. You should export these
+ * for other plugins to use in _their_ `SetupDeps`/`StartDeps` interfaces.
+ * @public
+ */
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface MapsLegacyLicensingSetupDependencies {}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface MapsLegacyLicensingStartDependencies {}
+
+export type MapsLegacyLicensingSetup = ReturnType<MapsLegacyLicensing['setup']>;
+export type MapsLegacyLicensingStart = ReturnType<MapsLegacyLicensing['start']>;
+
+export class MapsLegacyLicensing
+  implements Plugin<MapsLegacyLicensingSetup, MapsLegacyLicensingStart> {
+  public setup(core: CoreSetup, plugins: MapsLegacyLicensingSetupDependencies) {
+    const {
+      licensing,
+      mapsLegacy: { serviceSettings },
+    } = plugins;
+    if (licensing) {
+      licensing.license$.subscribe(({ uid, isActive }: { uid: string; isActive: boolean }) => {
+        if (isActive) {
+          serviceSettings.addQueryParams({ license: uid });
+          serviceSettings.disableZoomMessage();
+        } else {
+          serviceSettings.addQueryParams({ license: undefined });
+          serviceSettings.enableZoomMessage();
```

Comment:
thanks for introducing this. So basically, now it handles the case where the license becomes invalid after Kibana has started up?

Reply:
It does! I don't believe this was accounted for before. Much easier to do now with license status being observable. I also followed `addQueryParams` back through to `ems-client`. It ultimately functions more like a function named `setQueryParams` and should overwrite the previous license key/value. Having said that, I left this as a draft for now pending some final verification